inframon/tools/enclosure_api.py

```diff
@@ -0,0 +1,371 @@
+                            break
+                elif component_name == AlertConsts.NW_PARAMS:
+                    location[AlertConsts.CONTROLLER_ID] = an_element[AlertConsts.OBJECT_NAME]
+                else:
+                    if Consts.LOCATION in an_element:
+                        location[Consts.LOCATION_STR] = an_element[AlertConsts.LOCATION]
+                    else:
+                        location[Consts.LOCATION_STR] = entry
+
+    def get_status_raw(self, comp_type, **kwargs) -> Any:
+        """ Provides raw output of /api/show endpoints """
+        try:
+            uri, session_key = self._api_connection.get_session_key()
+            endpoint = EndpointBuilder.build(comp_type, **kwargs)
+            uri += endpoint
+            headers = {Consts.SESSION_KEY: session_key, Consts.DATA_TYPE: Consts.JSON}
+            r = requests.get(uri, headers=headers, verify=False)
+            response = json.loads(r.content)
+
+            return response
+        except EnclosureAPIError as exp:
+            raise EnclosureAPIError(ErrorMessages.ENC_RESP_ERROR) from exp
+
+    def get_status(self, comp_type, **kwargs) -> List[ToolResponse]:
+        """ Gets the information from /api/show endpoints
+            and converts the information to Json Response agreed with component monitor layer.
+        """
+        try:
+            response = self.get_status_raw(comp_type, **kwargs)
+            response = response[comp_type[Consts.RESPONSE]]
+            result = ResponseBuilderFactory().get(comp_type[Consts.NAME]).build_response(response, **kwargs)
+            to_delete = []
+            for i, a_result in enumerate(result):
+                resource_id = a_result.resource.resource_id
+                if resource_id in self._location_map:
+                    a_result.resource.location = self._location_map[resource_id]
+                else:
+                    # Remove these alerts for now. Add support later - supercap, system, snfs.
+                    to_delete.append(i)
+                    print(f"Error: Resource Id = {resource_id} not found.")
+
+            for count, i in enumerate(to_delete):
+                result.pop(i-count)
+
+            return result
+        except EnclosureAPIError as exp:
+            raise EnclosureAPIError(ErrorMessages.ENC_RESP_ERROR) from exp
+
+    def acknowledge(self, comp_type, alert_id) -> bool:
+        """Acknowledges an alert. An acknowledged alert can be filtered out for
+           identifying any new alert being generated in the system.
+        """
+        if comp_type[Consts.SHOW] == AlertConsts.ALERTS:
+            try:
+                uri, session_key = self._api_connection.get_session_key()
+                uri += Consts.URI_SET_ACK + str(alert_id)
+                headers = {Consts.SESSION_KEY: session_key, Consts.DATA_TYPE: Consts.JSON}
+                r = requests.get(uri, headers=headers, verify=False)
```

Comment:
Line 146 and 147 can be combined as `r` is not used further.

Reply:
This will look like -
response = json.loads(requests.get(uri, headers=headers, verify=False).content)
Makes a bit cryptic.